Installation issue: osu-micro-benchmarks | cannot link with -lcuda

### Steps to reproduce the issue

```console
$ spack spec -l osu-micro-benchmarks+cuda+xccl+graphing cuda_arch=80,89 %gcc@12.5.0 ^cuda@12.2.2
 -   7swvmtn  osu-micro-benchmarks@7.5.1+cuda+graphing~papi~rocm+xccl build_system=autotools cuda_arch:=80,89 platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  eiy76kp      ^compiler-wrapper@1.0 build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[+]  tr4udv7      ^cuda@12.2.2~allow-unsupported-compilers+dev build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[+]  bqvcwsc          ^libxml2@2.13.5~http+pic~python+shared build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  2lsfyjp              ^xz@5.6.3~pic build_system=autotools libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  a7o54p7      ^gcc@12.5.0+binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  xy6t6pc          ^binutils@2.45~debuginfod+gas~gold~gprofng~headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  ftx5kjo          ^diffutils@3.12 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  e4g27eg          ^gawk@5.3.1~nls build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  cxvj2r3              ^libsigsegv@2.14 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[e]  mhonfta          ^gcc@8.5.0~binutils+bootstrap~graphite~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' platform=linux os=rocky8 target=x86_64 
[+]  elvknsb          ^gcc-runtime@8.5.0 build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[+]  rfij6r7          ^gmp@6.3.0+cxx build_system=autotools libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  nfths67              ^m4@1.4.20+sigsegv build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  ek3pq55          ^libtool@2.4.7 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  tultd75              ^findutils@4.10.0 build_system=autotools patches:=440b954 platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  nwmfb3w          ^mpc@1.3.1 build_system=autotools libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  545ywq5          ^mpfr@4.2.1 build_system=autotools libs:=shared,static patches:=3ec29a6 platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  bpojoxd              ^autoconf-archive@2023.02.20 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 
[+]  bh6h47a          ^perl@5.42.0+cpanm+opcode+open+shared+threads build_system=generic platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  pxakavs              ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  olu4fsn              ^bzip2@1.0.8~debug~pic+shared build_system=generic platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  a7s773p              ^gdbm@1.25 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  bg4xidt          ^texinfo@7.2~xs build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  g5rd5kn              ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  gtbaqcm                  ^tar@1.35 build_system=autotools zip=pigz platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  frlfvym                      ^pigz@2.8 build_system=makefile platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  edfl5xk              ^ncurses@6.5-20250705~symlinks+termlib abi=none build_system=autotools patches:=7a351bc platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  oiyusni          ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  amjh65v          ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  voe4xrz      ^gcc-runtime@12.5.0 build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[e]  vsfl6cv      ^glibc@2.28 build_system=autotools platform=linux os=rocky8 target=x86_64 
[+]  4w4bokb      ^gmake@4.4.1~guile build_system=generic platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  en6mik6      ^gnuplot@6.0.0~X+cairo+gd+libcerf~pbm~qt+readline~wx build_system=autotools patches:=ad89f23 platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  fvthjml          ^cairo@1.18.4~X~dwrite+fc+ft~gobject+pdf+png~quartz~strip+svg~tee+zlib build_system=meson buildtype=release default_library:=shared platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  ldv6vl6              ^glib@2.86.1+introspection~libmount~strip build_system=meson buildtype=release default_library:=shared tracing:=none platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  y4bwkzt                  ^elfutils@0.193~debuginfod+exeprefix+nls build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  sfnvky3                  ^pcre2@10.44~jit+multibyte+pic build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  lafif3l              ^lzo@2.10 build_system=autotools libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  rspnz6q              ^meson@1.8.5 build_system=python_pip patches:=0f0b1bd platform=linux os=rocky8 target=x86_64_v3 
[+]  g6pzarq                  ^py-pip@25.1.1 build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[+]  bu6xgf4                  ^py-setuptools@80.9.0 build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[+]  hqqd4fu                  ^py-wheel@0.45.1 build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[+]  yjvmsvw              ^ninja@1.13.0+re2c build_system=generic platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  eiv4mlx                  ^re2c@3.1 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  7tlflvx              ^pixman@0.46.4+shared~strip build_system=meson buildtype=release default_library:=shared platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  ok2t6mz                  ^bison@3.8.2~color build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  nmblmve          ^libcerf@2.4~cpp~ipo build_system=cmake build_type=Release generator=make platform=linux os=rocky8 target=x86_64_v3 %c,fortran=gcc@12.5.0
[+]  4ecznaw              ^cmake@3.31.9~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  k5wekmq                  ^curl@8.15.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  h4dwnkq                      ^nghttp2@1.67.1 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  ofltznp          ^libgd@2.3.3 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  pzsrdsk              ^libx11@1.8.12 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  yc3yrfl                  ^inputproto@2.3.2 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  kxiu657                  ^kbproto@1.0.7 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  esanvqj                  ^libxcb@1.17.0 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  56dkynu                      ^libpthread-stubs@0.5 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  iu5rk6z                      ^libxau@1.0.12 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  sm3oror                      ^libxdmcp@1.1.5 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  e7mojdo                      ^xcb-proto@1.17.0 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 
[+]  rrw26fd                  ^xextproto@7.3.0 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  lsk3lf6                  ^xtrans@1.6.0 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  vu6462b          ^libiconv@1.18 build_system=autotools libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  m7ecyuh          ^libxpm@3.5.17 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  yllqgme              ^util-macros@1.20.1 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 
[+]  6eflqu2              ^xproto@7.0.31 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  2tjgskv          ^pango@1.54.0~X~strip build_system=meson buildtype=release default_library:=shared platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  4is4aup              ^fribidi@1.0.12 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  fwtsmfb              ^gobject-introspection@1.86.0~strip build_system=meson buildtype=release default_library:=shared patches:=96b2701 platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  opnf463                  ^glib-bootstrap@2.86.1~strip build_system=meson buildtype=release default_library:=shared platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  drn2mze              ^harfbuzz@11.5.1~graphite2~strip build_system=meson buildtype=release default_library:=shared platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  culxu7i                  ^icu4c@76.1 build_system=autotools cxxstd=17 platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  crvix2r                      ^python@3.14.2+bz2+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib+zstd build_system=generic platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  hlh4mwf                          ^expat@2.7.3+libbsd build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  n6fdwp7                              ^libbsd@0.12.2 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  23q6mj3                                  ^libmd@1.1.0 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  l5ji5uq                          ^libffi@3.5.2 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  hmlk66g                          ^sqlite@3.50.4+column_metadata+fts+rtree build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  zalv6vb                          ^util-linux-uuid@2.41 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  fu6ksko              ^json-glib@1.10.8~strip build_system=meson buildtype=release default_library:=shared platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  75i2hva              ^libffi@3.5.2 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  ecy7rmm          ^pkgconf@2.5.1 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  j2txl7t          ^readline@8.3 build_system=autotools patches:=21f0a03 platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  fzbsgbq      ^imagemagick@7.1.1-39~ghostscript~rsvg build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  uvgd4ju          ^fontconfig@2.15.0~pic build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,fortran=gcc@12.5.0
[+]  s2v237w              ^font-util@1.4.1 build_system=autotools fonts:=encodings,font-adobe-100dpi,font-adobe-75dpi,font-adobe-utopia-100dpi,font-adobe-utopia-75dpi,font-adobe-utopia-type1,font-alias,font-arabic-misc,font-bh-100dpi,font-bh-75dpi,font-bh-lucidatypewriter-100dpi,font-bh-lucidatypewriter-75dpi,font-bh-type1,font-bitstream-100dpi,font-bitstream-75dpi,font-bitstream-speedo,font-bitstream-type1,font-cronyx-cyrillic,font-cursor-misc,font-daewoo-misc,font-dec-misc,font-ibm-type1,font-isas-misc,font-jis-misc,font-micro-misc,font-misc-cyrillic,font-misc-ethiopic,font-misc-meltho,font-misc-misc,font-mutt-misc,font-schumacher-misc,font-screen-cyrillic,font-sun-misc,font-winitzki-cyrillic,font-xfree86-type1 platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  vejypia                  ^bdftopcf@1.1.1 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  yppx2mb                      ^fontsproto@2.1.3 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  kreigas                      ^libxfont@1.5.4 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  u3crjyf                  ^mkfontdir@1.0.7 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  2z7ss3o                  ^mkfontscale@1.2.3 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  tdixdba                      ^libfontenc@1.1.8 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  7nbnyws              ^gperf@3.1 build_system=autotools patches:=3dd36db platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  5hyk722              ^python@3.14.0+bz2+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib+zstd build_system=generic platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  qj25jjl                  ^expat@2.7.3+libbsd build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  rtny7yb                      ^libbsd@0.12.2 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  22e6s6n                          ^libmd@1.1.0 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  g2yucl6                  ^sqlite@3.50.4+column_metadata+fts+rtree build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  t57sxvw              ^util-linux-uuid@2.41 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  d5frpjc          ^freetype@2.14.1+pic+shared build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  eqccuzi          ^libjpeg-turbo@3.0.4~ipo~jpeg8~partial_decoder+pic build_system=cmake build_type=Release generator=make libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  prgpxxo              ^nasm@2.16.03 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  g5op7kb          ^libpng@1.6.47~ipo~pic build_system=cmake build_type=Release generator=make libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  ugucezt          ^libtiff@4.7.0+ccitt~ipo~jbig+jpeg~jpeg12~lerc~libdeflate+logluv~lzma+lzw+next~old-jpeg~opengl+packbits~pic~pixarlog+shared+thunder~webp+zlib~zstd build_system=cmake build_type=Release generator=make platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  5nkrz3s      ^nccl@2.28.9-1+cuda build_system=makefile cuda_arch:=80,89 fabrics:=verbs platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  mopu4kb          ^rdma-core@59.0~ipo+man_pages+pyverbs+static build_system=cmake build_type=Release generator=make platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  5wdj2zo              ^libnl@3.3.0 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  qgr6tx7              ^py-cython@3.1.3 build_system=python_pip platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  umynfke              ^py-docutils@0.21.2 build_system=python_pip platform=linux os=rocky8 target=x86_64_v3 
[+]  q3uehao                  ^py-flit-core@3.12.0 build_system=python_pip platform=linux os=rocky8 target=x86_64_v3 
[+]  pvhyjsc              ^python-venv@1.0 build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[+]  vfai6x6      ^openmpi@5.0.9+atomics+cuda~debug+fortran~gpfs~internal-hwloc~internal-libevent~internal-pmix~ipv6~java~lustre~memchecker~openshmem~rocm~romio+rsh~static~two_level_namespace+vt+wrapper-rpath build_system=autotools cuda_arch:=80,89 fabrics:=none romio-filesystem:=none schedulers:=none platform=linux os=rocky8 target=x86_64_v3 %c,cxx,fortran=gcc@12.5.0
[+]  47jl5fh          ^autoconf@2.72 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 
[+]  qj5wlmv          ^automake@1.16.5 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  arrzcll          ^hwloc@2.12.2+cairo+cuda~gl~level_zero~libudev+libxml2+nvml+opencl+pci~rocm build_system=autotools cuda_arch:=80,89 libs:=shared,static platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  lujquva              ^libpciaccess@0.17 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  d2u67rz          ^libevent@2.1.12+openssl build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  zscldlo              ^openssl@3.6.0~docs+shared build_system=generic certs=mozilla platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@8.5.0
[+]  pzglszn                  ^ca-certificates-mozilla@2025-08-12 build_system=generic platform=linux os=rocky8 target=x86_64_v3 
[+]  g6qdyv5          ^numactl@2.0.18 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  msrs6ve          ^openssh@9.9p1+gssapi build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  u62xrx6              ^krb5@1.21.3+shared build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0
[+]  h2mlp6a              ^libedit@3.1-20240808 build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  3wzdxbz              ^libxcrypt@4.4.38~obsolete_api build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@8.5.0
[+]  wlapj2m          ^pmix@6.0.0~munge~python build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  yg55qui          ^prrte@4.0.0 build_system=autotools schedulers:=none platform=linux os=rocky8 target=x86_64_v3 %c=gcc@12.5.0
[+]  remrev3              ^flex@2.6.3+lex~nls build_system=autotools platform=linux os=rocky8 target=x86_64_v3 %c,cxx=gcc@12.5.0

```


### Error message

<details><summary>Error message</summary>

<pre>
==> Error: ProcessError: Command exited with status 1:
    '/tmp/webf/spack-stage/spack-stage-osu-micro-benchmarks-7.5.1-7swvmtnafswyjzvo622jqkkgydk5ljb6/spack-src/configure' '--prefix=/cfs/earth/scratch/webf/opt/spack/opt/spack/linux-x86_64_v3/osu-micro-benchmarks-7.5.1-7swvmtnafswyjzvo622jqkkgydk5ljb6' 'CC=/cfs/earth/scratch/webf/opt/spack/opt/spack/linux-x86_64_v3/openmpi-5.0.9-vfai6x6ha2bwf4i5rdpq6ap4tca66372/bin/mpicc' 'CXX=/cfs/earth/scratch/webf/opt/spack/opt/spack/linux-x86_64_v3/openmpi-5.0.9-vfai6x6ha2bwf4i5rdpq6ap4tca66372/bin/mpic++' '--enable-cuda' '--with-cuda=/cfs/earth/scratch/webf/opt/spack/opt/spack/linux-x86_64_v3/cuda-12.2.2-tr4udv7y7csnmda5itenixm5aiksx6xh' '--enable-ncclomb' '--with-nccl=/cfs/earth/scratch/webf/opt/spack/opt/spack/linux-x86_64_v3/cuda-12.2.2-tr4udv7y7csnmda5itenixm5aiksx6xh' 'NVCCFLAGS=--generate-code arch=compute_80,code=sm_80 --generate-code arch=compute_80,code=compute_80 --generate-code arch=compute_89,code=sm_89 --generate-code arch=compute_89,code=compute_89' '--with-convert=/cfs/earth/scratch/webf/opt/spack/opt/spack/linux-x86_64_v3/imagemagick-7.1.1-39-fzbsgbq52duf5x5jggtosa2f3aqteexu/bin' '--with-gnuplot=/cfs/earth/scratch/webf/opt/spack/opt/spack/linux-x86_64_v3/gnuplot-6.0.0-en6mik6n5qgzmgjchozhobhem5ma5hdt/bin' 'LDFLAGS=-lrt'

1 error found in build log:
     134    checking for nccl.h... yes
     135    checking for library containing ncclGetUniqueId... -lnccl
     136    checking for ncclSend... yes
     137    checking for MPI_Session_init... (cached) yes
     138    checking for MPI_Psend_init... yes
     139    checking for library containing cuPointerGetAttribute... no
  >> 140    configure: error: cannot link with -lcuda
</pre></details>


### Information on your system

* **Spack:** 1.2.0.dev0 (https://github.com/spack/spack/commit/d9071ea6be97ea9976bce1eace84db255703777e)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/94f8eb2dcb1ffc989e24dd6f79bab0f193a2e9b0
* **Python:** 3.6.8
* **Platform:** linux-rocky8-zen2

### Additional information

[config.log](https://github.com/user-attachments/files/24165049/config.log)
[spack-build-env.txt](https://github.com/user-attachments/files/24165050/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/24165048/spack-build-out.txt)

@MatthewLieber  @harisubramoni  @natshineman

cuda related issue?: @Rombur  @ax3l  @pauleonix

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/osu_micro_benchmarks/package.py
+++ b/repos/spack_repo/builtin/packages/osu_micro_benchmarks/package.py
@@ -69,17 +69,17 @@ class OsuMicroBenchmarks(AutotoolsPackage, CudaPackage, ROCmPackage):
         config_args = ["CC=%s" % spec["mpi"].mpicc, "CXX=%s" % spec["mpi"].mpicxx]
 
         if "+cuda" in spec:
-            config_args.extend(["--enable-cuda", "--with-cuda=%s" % spec["cuda"].prefix])
+            config_args.extend(["--enable-cuda","--with-cuda=%s" % spec["cuda"].prefix])
             if spec.satisfies("+xccl"):
-                config_args.extend(["--enable-ncclomb", "--with-nccl=%s" % spec["cuda"].prefix])
+                config_args.extend(["--enable-ncclomb", "--with-nccl=%s" % spec["nccl"].prefix])
             cuda_arch = spec.variants["cuda_arch"].value
             if "none" not in cuda_arch:
                 config_args.append("NVCCFLAGS=" + " ".join(self.cuda_flags(cuda_arch)))
 
         if "+rocm" in spec:
-            config_args.extend(["--enable-rocm", "--with-rocm=%s" % spec["hip"].prefix])
+            config_args.extend([ "--with-rocm=%s" % spec["hip"].prefix])
             if spec.satisfies("+xccl"):
-                config_args.extend(["--enable-rcclomb", "--with-rccl=%s" % spec["hip"].prefix])
+                config_args.extend(["--enable-rcclomb", "--with-rccl=%s" % spec["rccl"].prefix])
             rocm_arch = spec.variants["amdgpu_target"].value
             if "none" not in rocm_arch:
                 config_args.append("HCC_AMDGPU_TARGET=" + self.hip_flags(rocm_arch))
@@ -98,6 +98,9 @@ class OsuMicroBenchmarks(AutotoolsPackage, CudaPackage, ROCmPackage):
         if not sys.platform == "darwin":
             config_args.append("LDFLAGS=-lrt")
         return config_args
+    def setup_build_environment(self, env: EnvironmentModifications) -> None:
+        if "+cuda" in self.spec:
+            env.prepend_path('LD_LIBRARY_PATH',self.spec["cuda"].prefix.lib64.stubs)
 
     def setup_run_environment(self, env: EnvironmentModifications) -> None:
         mpidir = join_path(self.prefix.libexec, "osu-micro-benchmarks", "mpi")

--- a/repos/spack_repo/builtin/packages/osu_micro_benchmarks/package.py
+++ b/repos/spack_repo/builtin/packages/osu_micro_benchmarks/package.py
@@ -69,7 +69,7 @@ class OsuMicroBenchmarks(AutotoolsPackage, CudaPackage, ROCmPackage):
         config_args = ["CC=%s" % spec["mpi"].mpicc, "CXX=%s" % spec["mpi"].mpicxx]
 
         if "+cuda" in spec:
-            config_args.extend(["--enable-cuda","--with-cuda=%s" % spec["cuda"].prefix])
+            config_args.extend(["--enable-cuda", "--with-cuda=%s" % spec["cuda"].prefix])
             if spec.satisfies("+xccl"):
                 config_args.extend(["--enable-ncclomb", "--with-nccl=%s" % spec["nccl"].prefix])
             cuda_arch = spec.variants["cuda_arch"].value
@@ -77,7 +77,7 @@ class OsuMicroBenchmarks(AutotoolsPackage, CudaPackage, ROCmPackage):
                 config_args.append("NVCCFLAGS=" + " ".join(self.cuda_flags(cuda_arch)))
 
         if "+rocm" in spec:
-            config_args.extend([ "--with-rocm=%s" % spec["hip"].prefix])
+            config_args.extend(["--with-rocm=%s" % spec["hip"].prefix])
             if spec.satisfies("+xccl"):
                 config_args.extend(["--enable-rcclomb", "--with-rccl=%s" % spec["rccl"].prefix])
             rocm_arch = spec.variants["amdgpu_target"].value
@@ -98,9 +98,10 @@ class OsuMicroBenchmarks(AutotoolsPackage, CudaPackage, ROCmPackage):
         if not sys.platform == "darwin":
             config_args.append("LDFLAGS=-lrt")
         return config_args
+
     def setup_build_environment(self, env: EnvironmentModifications) -> None:
         if "+cuda" in self.spec:
-            env.prepend_path('LD_LIBRARY_PATH',self.spec["cuda"].prefix.lib64.stubs)
+            env.prepend_path("LD_LIBRARY_PATH", self.spec["cuda"].prefix.lib64.stubs)
 
     def setup_run_environment(self, env: EnvironmentModifications) -> None:
         mpidir = join_path(self.prefix.libexec, "osu-micro-benchmarks", "mpi")

